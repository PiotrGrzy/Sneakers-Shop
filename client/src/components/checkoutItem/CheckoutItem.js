import React from 'react';
import { connect } from 'react-redux';
import {
  addItemToCart,
  decrementItem,
  removeItemFromCart
} from '../../redux/cart/cart.actions';

import './checkout-item.scss';

const CheckoutItem = ({
  item,
  addItemToCart,
  decrementItem,
  removeItemFromCart
}) => {
  const { imageMain, model, brand, quantity, price, size } = item;

  return (
    <div className="checkout-item">
      <img src={imageMain.url} alt="Sneaker" className="checkout-item__img" />
      <div className="checkout-item__brand">{brand}</div>
      <div className="checkout-item__model">{model}</div>
      <div className="checkout-item__size">{size}</div>
      <div className="checkout-item__minus" onClick={() => decrementItem(item)}>
        &#10094;
      </div>
      <div className="checkout-item__quantity">{quantity}</div>
      <div className="checkout-item__plus" onClick={() => addItemToCart(item)}>
        &#10095;
      </div>
      <div className="checkout-item__price">{price} $</div>
      <div
        className="checkout-item__delete"
        onClick={() => removeItemFromCart(item)}
      >
        &#10005;
      </div>
    </div>
  );
};

export default connect(null, {
  addItemToCart,
  decrementItem,
  removeItemFromCart
})(CheckoutItem);
