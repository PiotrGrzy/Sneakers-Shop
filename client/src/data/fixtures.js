export const item = {
  id: 'ck75wcleo155b0b843rlkt044',
  price: 129,
  gender: 'Men',
  category: ['Running'],
  imageMain: {
    id: 'ck75w55qoe0v90b201x2oqo8l',
    url: 'https://media.graphcms.com/JqjJAtRTlethzLQxhUtQ',
  },
  model: 'Zoom 210',
  imageSecondary: {
    id: 'ck75w55kfe0v40b209n4moxvp',
    url: 'https://media.graphcms.com/EVQOXidUTZksHn7HzJAQ',
  },
  sizes: '40 41 42 43 44 45 46 47',
  brand: 'Nike',
  trending: true,
};
