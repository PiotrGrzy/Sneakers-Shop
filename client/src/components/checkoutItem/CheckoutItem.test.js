import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { CheckoutItem } from './CheckoutItem';
import { item } from '../../data/fixtures';

const props = {
  item: { ...item, quantity: 1 },
  addItemToCart: jest.fn(),
  decrementItem: jest.fn(),
  removeItemFromCart: jest.fn(),
};

describe('CheckoutItem', () => {
  const wrapper = shallow(<CheckoutItem {...props} />);
  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
