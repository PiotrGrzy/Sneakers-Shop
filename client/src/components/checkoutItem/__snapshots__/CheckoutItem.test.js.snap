// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem renders properly 1`] = `ShallowWrapper {}`;
